packages/builds/src/node/error-parser.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as Events from "events";
+import * as path from "path";
+import { injectable, decorate } from "inversify";
+
+export const IErrorParser = Symbol("IErrorParser");
+
+export interface IErrorParser extends Events.EventEmitter {
+    /**
+     * starts the parsing, returns a list of errors/warnings found in the input,
+     * matching the given the error matcher.
+     */
+    parse(errorMatcher: IErrorMatcher, inputStream: NodeJS.ReadableStream): Promise<IParsedError[]>
+}
+
+/**
+ * A parsed error or warning
+ */
+export interface IParsedError {
+    /** whether this entry is an error, a warning */
+    severity: string,
+    /** name of the file where the error occurs */
+    file: string,
+    /** location within the file. e.g. source code line number, column */
+    location: string,
+    /** text of the error or warning */
+    message: string,
+    /** reported error code, if any */
+    code: string
+}
+
+export interface IErrorMatcher {
+    // TODO: not all fields are yet used...
+    "name": string,
+    "label": string,
+    /** maps to the build command type? e.g. "typescript", "gcc/g++", ... */
+    "owner": string,
+    /** paths generated are absolute or relative? values: 'absolute' or 'relative' */
+    "fileLocation": string,
+    /** Regexp that matches errors/warnings */
```

Comment:
Sorry that was not very clear. should be better now. BTW name borrowed from vscode